backpack/extensions/module_extension.py

```diff
@@ -127,25 +127,40 @@ def __call__(
                 extValue = extFunc(extension, module, g_inp, g_out, bp_quantity)
                 self.__save_value_on_parameter(extValue, extension, module, param)
 
-        if self.__should_backpropagate(extension, module):
+        module_inputs = self.__should_backpropagate(extension, module)
+        if module_inputs:
             bp_quantity = self.backpropagate(
                 extension, module, g_inp, g_out, bp_quantity
             )
-            self.__save_backproped_quantity(extension, module.input0, bp_quantity)
+            for module_inp in module_inputs:
+                self.__save_backproped_quantity(extension, module_inp, bp_quantity)
 
     @staticmethod
-    def __should_backpropagate(extension: BackpropExtension, module: Module) -> bool:
-        """Determines whether the current extension should perform a backpropagation.
+    def __should_backpropagate(
```

Comment:
Suggestion: Rename into `__backpropagate_to`?

Reply:
Yes, the name should be changed.
`__backpropagate_to` suggests that the backpropagation is happening inside the function.
Instead I chose `__get_inputs_for_backpropagation`. However, it is a bit lengthy. Do you have something better?